Library/Homebrew/env_config.rb

```diff
@@ -96,6 +96,10 @@ module EnvConfig
                       "Linux: `https://github.com/Homebrew/linuxbrew-core`.",
         default:      HOMEBREW_CORE_DEFAULT_GIT_REMOTE,
       },
+      HOMEBREW_CURL_PATH:                     {
+        description: "Use this executable when invoking `curl`(1).",
```

Comment:
```suggestion
        description: "Use this executable when invoking `curl`(1). Incompatible with `HOMEBREW_FORCE_BREWED_CURL`.",
```

and same with `HOMEBREW_FORCE_BREWED_CURL`; note `HOMEBREW_CURL_PATH` is incompatible.